src/app/RegionCodeSearch.tsx

```diff
@@ -1,23 +1,25 @@
 "use client";
 
 import { FormEvent, useState } from "react";
-import { findAllRegions, Region } from "./actions";
+import { findMatchingRegions } from "./actions";
 import Link from "next/link";
 import { Button } from "@/components/ui/button";
 import { Input } from "@/components/ui/input";
 import { Label } from "@/components/ui/label";
+import { FuseResult } from "fuse.js";
+import { FlattenedRegion } from "./actions";
 
 export function RegionCodeSearch() {
   "use client";
   const [searchQuery, setSearchQuery] = useState("");
-  const [allMatchingRegions, setAllMatchingRegions] = useState<Array<Region>>(
-    [],
-  );
+  const [regionSearchResults, setRegionSearchResults] = useState<
+    FuseResult<FlattenedRegion>[]
+  >([]);
 
   const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
     event.preventDefault();
-    const regions = await findAllRegions(searchQuery);
-    setAllMatchingRegions(regions);
+    const fuseSearchMatches = await findMatchingRegions(searchQuery);
```

Comment:
One thing to be extra diligent of when responding to PR comments is that you are checking other places that might be worth applying the comment to. 

i.e here there's also a reference to `fuse` which we can probably remove as well. 

```suggestion
    const regionSearchMatches = await findMatchingRegions(searchQuery);
```

Reply:
Ah I noticed you commented on this in the original place. 

I've left a reply there to discuss 🙏